src/main/haskell/kore/src/Kore/SMT/SMT.hs

```diff
@@ -49,121 +68,169 @@ instance ParseAttributes SMTAttributes where
         hook <- attributesParser
         pure SMTAttributes { hook }
 
-provePredicate
-    :: Given (MetadataTools Object SMTAttributes)
-    => KorePredicate.CommonPredicate Object
+-- Either a variable with a SMT-hooked sort
+-- or a pattern with an Uninterpreted Function head
+type VarOrUF = Either (Id Object) Pat
+
+data TranslationState
+ = TranslationState
+   { _boolVars :: Map.Map VarOrUF SBool
+   , _intVars  :: Map.Map VarOrUF SInteger
+   }
+
+makeLenses ''TranslationState
+
+getVar
+    :: Lens' TranslationState (Map.Map VarOrUF b)
+    -> (String -> Symbolic b)
+    -> VarOrUF
+    -> Translating b
+getVar lens makeNewVar i = do
+  v' <- Map.lookup i <$> use lens
+  case v' of
+    Nothing -> do
+      v <- lift $ lift $ makeNewVar (either getId show i)
+      lens %= Map.insert i v
+      return v
+    Just v -> return v
+
+getBoolVar :: VarOrUF -> Translating SBool
+getBoolVar = getVar boolVars sBool
+
+getIntVar :: VarOrUF -> Translating SInteger
+getIntVar  = getVar intVars sInteger
+
+goUnaryOp
+    :: Monad m
+    => (t1 -> b)
+    -> (t2 -> m t1)
+    -> t2
+    -> m b
+goUnaryOp op cont x1 = do
+    tx1 <- cont x1
+    return (op tx1)
+
+goBinaryOp
+    :: Monad m
+    => (t1 -> t1 -> b)
+    -> (t2 -> m t1)
+    -> t2 -> t2
+    -> m b
+goBinaryOp op cont x1 x2 = do
+    tx1 <- cont x1
+    tx2 <- cont x2
+    return (tx1 `op` tx2)
+
+config :: SMTConfig
+config = z3 -- { transcript = Just "/Users/phillip/smt.log"}
+
+-- | Returns `Just False` if the SMT solver can prove the pattern
+-- is undecidable, and `Nothing` otherwise. 
+unsafeTryRefutePattern
+    :: ( Given (MetadataTools Object SMTAttributes)
+       , Ord (variable Object)
+       , SortedVariable variable
+       )
+    => PureMLPattern Object variable
     -> Maybe Bool
-provePredicate p = provePattern $ KorePredicate.unwrapPredicate p
+unsafeTryRefutePattern p = unsafePerformIO $ do
+  let smtPredicate = setTimeOut 20 >> patternToSMT True p -- 20ms 
```

Comment:
Space at the end of the line. Most editors allow you to automatically remove this kind of space, or, at least, to highlight it if, for whatever reason, you don't want the automatic stuff. Even if that doesn't work, it's likely that it's easy to write a small script to do that for an entire pull request (though it may require some fiddling to find out exactly what is included in the pull request)